### PR TITLE
Nagios: added 'acknowledge' and 'schedule_forced_svc_check' action

### PR DESCRIPTION
**Pull request by tgoetheyn**
##### SUMMARY

Adds the option "acknowledge" and 'schedule_forced_svc_check' to the module actions.
This way the module can be used to acknowledge open host & service problems and to force a recheck of passive services in nagios

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

nagios

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: changelogs/fragments/820_nagios_added_acknowledge_and_servicecheck.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+- nagios - add the ``acknowledge`` parameter (https://github.com/ansible-collections/community.general/pull/820).
+- nagios - add the ``service_check`` parameter (https://github.com/ansible-collections/community.general/pull/820).
```
Comment (felixfontein) [inline changelogs/fragments/820_nagios_added_acknowledge_and_servicecheck.yml:3]: These are not parameters, but actions, I think? From your change, it does not look like you are adding new module options (parameters).

Comment (tgoetheyn) [reply changelogs/fragments/820_nagios_added_acknowledge_and_servicecheck.yml:3]: you're right, my mistake.
I'll correct it right away

File: plugins/modules/monitoring/nagios.py [hunk 2]
```diff
@@ -22,21 +22,22 @@
...
+    I(Setting alerts/downtime/acknowledge for a host does not affect alerts/downtime/acknowledge for any of the services running on it.)
+    To schedule downtime for all services on particular host use keyword "all", e.g., C(service=all).
   - When using the C(nagios) module you will need to specify your Nagios server using the C(delegate_to) parameter.
 options:
   action:
     description:
       - Action to take.
       - servicegroup options were added in 2.0.
       - delete_downtime options were added in 2.2.
+      - acknowledge and service_check options were added in 2.10.
```
Comment (felixfontein) [inline plugins/modules/monitoring/nagios.py:36]:
```suggestion
      - The C(acknowledge) and C(service_check) actions were added in community.general 1.2.0.
```

**Review by felixfontein — COMMENTED** [review]
(I cannot comment on the code change itself.)

**Comment by tgoetheyn** [comment]
can someone please review this?

File: plugins/modules/monitoring/nagios.py [hunk 3]
```diff
@@ -358,7 +395,7 @@ def __init__(self, module, **kwargs):
         self.comment = kwargs['comment']
         self.host = kwargs['host']
         self.servicegroup = kwargs['servicegroup']
-        self.start = int(kwargs['start'])
+        self.start = kwargs['start']
```
Comment (felixfontein) [inline plugins/modules/monitoring/nagios.py:398]: Why did you remove the `int()`? This way, you end up with a string.

(The module's argument handling needs an overhaul anyway, most types are not specified. But that's for another PR...)

**Review by felixfontein — COMMENTED** [review]
Besides this, the PR looks like it won't break existing use-cases.

File: plugins/modules/monitoring/nagios.py (hunk 3 above)
Comment (tgoetheyn) [reply plugins/modules/monitoring/nagios.py:398]: because it is an optional parameter.
When it is not provided the module throws an 'cannot convert NullType' error

Comment (felixfontein) [reply plugins/modules/monitoring/nagios.py:398]: But in that case, you need to only use `int()` if it is not `None`. Completely removing it can potentially break parts of the module which rely on this being an int.

File: plugins/modules/monitoring/nagios.py [hunk 4]
```diff
@@ -395,7 +395,10 @@ def __init__(self, module, **kwargs):
         self.comment = kwargs['comment']
         self.host = kwargs['host']
         self.servicegroup = kwargs['servicegroup']
-        self.start = kwargs['start']
+       if not kwargs['start'] == None:
```
Comment (felixfontein) [inline plugins/modules/monitoring/nagios.py:398]:
```suggestion
       if kwargs['start'] is not None:
```

**Comment by tgoetheyn** [comment]
Hi @felixfontein, thanks for the guidance!
I fixed the 'int' convertion issue, so hopefully the pull request can be merged now.

Additionally I was wondering if it is possible to remove @tbielawa as a maintainer as I never got any respone from him since my [initial PR](https://github.com/ansible/ansible/pull/49767) back in 2018.
I don't know if it is possible to assign me as an maintainer because i'm still a new contributer?

**Review by felixfontein — APPROVED** [review]
Looks like it doesn't break anything. Cannot comment on whether it does what it promises.

**Comment by felixfontein** [comment]
> Hi @felixfontein, thanks for the guidance!
> I fixed the 'int' convertion issue, so hopefully the pull request can be merged now.

I think it looks fine. I'll keep an eye on it, if nobody complains I'll merge it beinning of next week so it will be included in community.general 1.2.0.

> Additionally I was wondering if it is possible to remove @tbielawa as a maintainer as I never got any respone from him since my [initial PR](https://github.com/ansible/ansible/pull/49767) back in 2018.
> I don't know if it is possible to assign me as an maintainer because i'm still a new contributer?

I'm not sure about when we remove existing maintainers, but we can definitely add you! Would be great if you're interested in improving the module and helping with issues! (A great start would be fixing all the ignored sanity errors: https://github.com/ansible-collections/community.general/blob/main/tests/sanity/ignore-2.11.txt#L680-L683)

**Comment by tgoetheyn** [comment]
> I'm not sure about when we remove existing maintainers, but we can definitely add you! Would be great if you're interested in improving the module and helping with issues! 

Sure! always happy to help

> A great start would be fixing all the ignored sanity errors: https://github.com/ansible-collections/community.general/blob/main/tests/sanity/ignore-2.11.txt#L680-L683)

I'll look in to it!

**Comment by felixfontein** [comment]
@tgoetheyn thanks a lot for working on this module!
